File: Triangles.py
import sys

sys.stdin = open("triangles.in", "r")
sys.stdout = open("triangles.out", "w")

N=(int(sys.stdin.readline()))

posts=[]
for _ in range(N):
    posts.append((list(map(int,sys.stdin.readline().split()))))


# max_area=0
# for point in posts:
#     x,y=point
#     max_x=0
#     max_y=0
#     for x0 in posts:
#         if x0!=point:
#             x_val=x0[0]
#             y_val=x0[1]
#             if y==y_val:
#                 if abs(x_val-x)>max_x:
#                     max_x=abs(x_val-x)         
#     for y0 in posts:
#         if y0!=point:
#             x_val=y0[0]
#             y_val=y0[1]
#             if x==x_val:
#                 if abs(y_val-y)>max_y:
#                     max_y=abs(y_val-y)
#     max_area=max(max_area,max_x*max_y)
# print(max_area)

max_area=0
for point in posts:
    x,y=point
    max_x=0
    max_y=0
    for two in posts:
        if two!=point:
            x_val=two[0]
            y_val=two[1]
            if x==x_val:
                max_y=max(max_y,abs(y_val-y))
            if y==y_val:
                max_x=max(max_x,abs(x_val-x))      
    max_area=max(max_area,max_x*max_y)
print(max_area)

File: Triangles.java
import java.io.*;
import java.util.StringTokenizer;
import java.lang.Math;

public class Triangles {
	public static void main(String[] args) throws IOException {
		BufferedReader r = new BufferedReader(new FileReader("triangles.in"));
		PrintWriter pw = new PrintWriter(new BufferedWriter(new FileWriter("triangles.out")));
        StringTokenizer st = new StringTokenizer(r.readLine());
        int N = Integer.parseInt(st.nextToken());
        int[] left=new int[N];
        int[] right=new int[N];
        for (int i=0; i<N; i++){
            st = new StringTokenizer(r.readLine());
            left[i]=Integer.parseInt(st.nextToken());
            right[i]=Integer.parseInt(st.nextToken());
        }
        int max_area = 0;
        for (int i=0; i<N; i++){
            int x=left[i];
            int y=right[i];
            int max_x=0;
            int max_y=0;
            for (int j=0; j<N; j++){
                int x_val=left[j];
                int y_val=right[j];
                if (x_val!=x || y_val!=y){
                    if (x==x_val){
                        max_y = Math.max(max_y,Math.abs(y_val-y));
                    }
                    if (y==y_val){
                        max_x = Math.max(max_x,Math.abs(x_val-x));
                    }
                }
            }
            max_area = Math.max(max_area,max_x*max_y);
        }

		pw.println(max_area);

		pw.close();
	}
}
